// vitepress-theme-teek/es/components/theme/ArticleTitle/src/index.vue2.mjs
import { defineComponent, createElementBlock, openBlock, unref, normalizeClass, createBlock, createCommentVNode, renderSlot, mergeProps, Fragment, resolveDynamicComponent, toDisplayString } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import '../../../../composables/useZIndex.mjs';
import _sfc_main$1 from '../../../common/TitleTag/src/index.vue2.mjs';
import { createDynamicComponent } from './compile.mjs';

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = { key: 1 };
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "ArticleTitle" },
  __name: "index",
  props: {
    post: {},
    titleTagProps: {}
  },
  setup(__props) {
    const ns = useNamespace("article-title");
    const { t } = useLocale();
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("span", {
        class: normalizeClass(unref(ns).b()),
        "aria-label": unref(t)("tk.articleTitle.label")
      }, [
        __props.post.frontmatter.titleTag && __props.titleTagProps?.position === "left" ? (openBlock(), createBlock(unref(_sfc_main$1), mergeProps({
          key: 0,
          text: __props.post.frontmatter.titleTag
        }, __props.titleTagProps, {
          "aria-label": __props.post.frontmatter.titleTag
        }), null, 16, ["text", "aria-label"])) : createCommentVNode("v-if", true),
        renderSlot(_ctx.$slots, "default", {}, () => [
          __props.post.title ? (openBlock(), createElementBlock(
            Fragment,
            { key: 0 },
            [
              !__props.post.title.includes("`<") && !__props.post.title.includes(">`") ? (openBlock(), createBlock(resolveDynamicComponent(unref(createDynamicComponent)(__props.post.title)), { key: 0 })) : (openBlock(), createElementBlock(
                "span",
                _hoisted_2,
                toDisplayString(__props.post.title),
                1
                /* TEXT */
              ))
            ],
            64
            /* STABLE_FRAGMENT */
          )) : createCommentVNode("v-if", true)
        ]),
        __props.post.frontmatter.titleTag && __props.titleTagProps?.position === "right" ? (openBlock(), createBlock(unref(_sfc_main$1), mergeProps({
          key: 1,
          text: __props.post.frontmatter.titleTag
        }, __props.titleTagProps, {
          "aria-label": __props.post.frontmatter.titleTag
        }), null, 16, ["text", "aria-label"])) : createCommentVNode("v-if", true)
      ], 10, _hoisted_1);
    };
  }
});

export { _sfc_main as default };
